lib/filewatch/discover.rb

```diff
@@ -0,0 +1,141 @@
+require_relative 'watched_file'
+
+module FileWatch
+  class Discover
+    attr_reader  :logger, :files, :wf_vars
+
+    def initialize(opts, loggr)
+      @logger = loggr
+      @watching = []
+      @exclude = []
+      @files = {}
+      set_ignore_older(opts[:ignore_older])
+      set_close_older(opts[:close_older])
+      @wf_vars = [opts[:delimiter], @ignore_older, @close_older]
+      exclude(opts[:exclude])
+    end
+
+    def logger=(loggr)
+      @logger = loggr
+      @converter.logger = loggr
+    end
+
+    def add_converter(converter)
+      @converter = converter
+    end
+
+    def add_path(path)
+      return if @watching.member?(path)
+      @watching << path
+      discover_file(path) do |fpath, stat|
+        WatchedFile.new_initial(*wf_args(fpath, stat)).init_vars(*wf_vars)
+      end
+    end
+
+    def discover
+      @watching.each do |path|
+        discover_file(path) do |fpath, stat|
+          WatchedFile.new_ongoing(*wf_args(fpath, stat)).init_vars(*wf_vars)
+        end
+      end
+    end
+
+    def delete(paths)
+      paths.each {|f| @files.delete(f)}
+    end
+
+    def close_all
+      @files.values.each(&:file_close)
+    end
+
+    def empty?
+      @files.empty?
+    end
+
+    def watched_files
+      @files.values
+    end
+
+    private
+
+    def set_ignore_older(value)
+      #nil is allowed but 0 and negatives are made nil
+      if !value.nil?
+        val = value.to_f
+        val = val <= 0 ? nil : val
+      end
+      @ignore_older = val
+    end
+
+    def set_close_older(value)
+      if !value.nil?
+        val = value.to_f
+        val = val <= 0 ? nil : val
+      end
```

Comment:
line 63-65 and 71-74 are exactly the same code, might be worth it to create a coercion method for that?